netsim/ansible/tasks/deploy-config/eos.yml

```diff
@@ -1,4 +1,24 @@
+- set_fact: deployed_config={{ lookup('template',config_template) }}
+- set_fact: bash_script={{ "#!/bin/bash" in deployed_config }}
+
+- template:
+    src: "{{ config_template }}"
+    dest: /tmp/config.sh
+  when: bash_script
```

Comment:
You could group multiple tasks with the same **when** condition into a block. Never tried what happens when a task inside a **block** has another **when** condition though (and it might be Ansible release dependent)